MatrixSDK/Utils/Categories/FileManager+Backup.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+
+public extension FileManager {
+    /**
+     Create a directory at a url, including intermediate directories, which is excluded from iCloud / manual backups, no matter where it is located.
+     
+     Note: some directories are excluded automatically if they are nested within `<AppHome>/Library/Caches/` or `<AppHome>/tmp/`
+     see [details](https://developer.apple.com/documentation/foundation/optimizing_app_data_for_icloud_backup).
+     */
+    @objc func createDirectoryExcludedFromBackup(at url: URL) throws {
```

Comment:
I considered an alternative api along the lines of `createDirectory(at:withIntermediate:attributes:isExcluded:)` but we just pass the same parameters everywhere anyway, so it felt cumbersome and pointless.

Reply:
This api distinction already exists within `FileManager`'s API (`(at url: URL)` vs `(atPath path: String)`). Reading [swift conventions](https://www.swift.org/documentation/api-design-guidelines/#parameter-names) I assume that `(at url: URL)` is preferable, but then `(at path: String)` could not be easily distinguished / would be overloading. Since `URL` is preferable type over `String` path, the url-api is given preferential treatment. I don't mind aligning them with your suggestion, just wanted to point out that this was a deliberate choice given the existing `FileManager` api.

Reply:
If it were my choice I would make them symmetrical but no strong feelings really.

Reply:
Yea after I changed the `excludeItemFromBackup` method, it seems preferable to stick with `(at url: URL)` everywhere for consistency sake.